server/export.js

```diff
@@ -255,82 +263,176 @@ var draw = function(res, filename, map, x, y){
 };
 
 var thumbnail_draw = function(res, filename, map){
-	var x = 200;
-	var y = 200;
-	var canvas = new Canvas(x,y);
-	var stream = canvas.pngStream();
-	var ctx = canvas.getContext('2d');
-
-	// background
-	ctx.fillStyle = "#FFFFFF";
-	ctx.fillRect(0,0,x,y);
-
-	// calculate nodes
-
-	var availableWidth = x ;
-	var availableHeight = y ;
-
-	var calculatedNodes = [];
-	for(var index in map.nodes){
-		var node = map.nodes[index];
-		var cX = Math.floor(node.positionX * availableWidth);
-		var cY = Math.floor(node.positionY * availableHeight);
-		calculatedNodes.push({
-			x : cX,
-			y : cY,
-			name : node.name,
-			componentId : node.componentId
-		});
-	}
-
-	ctx.strokeStyle = 'silver';
-	ctx.fillStyle = 'silver';
-
-	//draw connections
-	for(var index in map.connections){
-		var connection = map.connections[index];
-		var source = connection.pageSourceId;
-		var target = connection.pageTargetId;
-		var start;
-		var end;
-		for(var nodeIndex in calculatedNodes){
-
-			if(calculatedNodes[nodeIndex].componentId == source){
-				start = calculatedNodes[nodeIndex];
-			}
-			if(calculatedNodes[nodeIndex].componentId == target){
-				end = calculatedNodes[nodeIndex];
+	var htmlStub = '<html><head><script></script></head><body><div id="map-container"></div></body></html>';
+	var scripts = ['//cdnjs.cloudflare.com/ajax/libs/d3/3.5.5/d3.js', '//cdnjs.cloudflare.com/ajax/libs/lodash.js/3.2.0/lodash.min.js'];
+	
+	jsdom.env({
+		features : {
+			QuerySelector : true,
+			FetchExternalResources : [ 'script', 'css' ],
+			ProcessExternalResources : [ 'script', 'css' ]
+		},
+		html : htmlStub,
+		scripts : scripts, 
+		done : function(errors, window) {
+			
+			function pick(key) {
+				return function(d) {
+					if(typeof d[key] === 'string' && d[key] === 'true'){
+						return true;
+					}
+					if(typeof d[key] === 'string' && d[key] === 'false'){
+						return false;
+					}
+					return d[key];
+				};
 			}
 
+			function unitify(unit) {
+				return function(d) {
+					return '' + d + unit;
+				};
+			}
+				
+			
+			var x = 1280;
+			var y = 1024;
+			var thumbnailMargin = 20;
+			
+			var el = window.document.querySelector('#map-container');
+			var svgimg = d3.select(el).append('svg:svg').attr('xmlns', 'http://www.w3.org/2000/svg').attr('class', 'map');
+			
+			
+			var defs = svgimg.append('defs');
+			
+			var css = defs.append('style').attr('type','txt/css');
```

Comment:
the `<style>` element should be a child of the `<svg>` element, not `<defs>`, then the css will work